### PR TITLE
fix: full clone of template repos on add

### LINKED ISSUE #891: Verbose flag for repository list command doesn't work
By reading https://github.com/knative-sandbox/kn-plugin-func/blob/main/docs/guides/commands.md#list-1 the flag `-v` or `--verbose` should also list the URLs of each repo but it only shows the names as if the flag it is not used

```
$ func repository list
default
myrepo
$ func repository list -v
default
myrepo
```

### PR DESCRIPTION
**Pull request by lkingland**
# Changes

- :bug: Repositories missing .git

Repositories added via `func repo add ...` were missing the .git directory when installed locally.  The intent of `func repo add` was to be functionally equivalent to a `git clone`, so missing the .git directory is a bug.  This is evidenced by the downstream bug that reading metadata, such as the source URL, failed due to the missing metadata directory.

Note that existing (installed) repositories will need to be removed and re-added to be correctly installed.

This fixes the aforementioned problem of a missing remote URL when listing repositories in verbose mode (#891)

/kind bug

Fixes #891

### CODE REVIEW

File: repository.go [hunk 1]
```diff
@@ -449,24 +450,53 @@ func (r *Repository) Runtime(name string) (runtime Runtime, err error) {
...
 }
 
 // URL attempts to read the remote git origin URL of the repository.  Best
 // effort; returns empty string if the repository is not a git repo or the repo
 // has been mutated beyond recognition on disk (ex: removing the origin remote)
 func (r *Repository) URL() string {
-	repo, err := git.PlainOpen(r.uri)
+	uri := r.uri
+
+	// git.PlainOpen only accepts paths
```
Comment (matejvasek) [inline repository.go:494]: @lkingland 
What if it's `http(s)://`?
The `git.PlainOpen()` would fail?

**Comment by matejvasek** [comment]
@lkingland this will print URL on Linux with the `-v` flag.
But on `wine` it won't print URL, and on actual Windows machine this even panics (it panicked even before this PR):
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x18 pc=0xc43e88]
goroutine 1 [running]:
github.com/go-git/go-git/v5.dotGitToOSFilesystems(0xc00004ed40, 0x32, 0xc0007c8000, 0x9, 0xd8, 0x9, 0x0, 0x0, 0x100000050305e)
        /home/mvasek/go/src/github.com/knative-sandbox/kn-plugin-func/vendor/github.com/go-git/go-git/v5/repository.go:283 +0x4c8
github.com/go-git/go-git/v5.PlainOpenWithOptions(0xc000b74486, 0x30, 0xc000d1face, 0x1, 0x0, 0x0)
        /home/mvasek/go/src/github.com/knative-sandbox/kn-plugin-func/vendor/github.com/go-git/go-git/v5/repository.go:241 +0x6e
github.com/go-git/go-git/v5.PlainOpen(...)
        /home/mvasek/go/src/github.com/knative-sandbox/kn-plugin-func/vendor/github.com/go-git/go-git/v5/repository.go:235
knative.dev/kn-plugin-func.(*Repository).URL(0xc000d1fb88, 0xc000006018, 0xc000d1fb78)
        /home/mvasek/go/src/github.com/knative-sandbox/kn-plugin-func/repository.go:499 +0xa8
knative.dev/kn-plugin-func/cmd.runRepositoryList(0xc0008771b0, 0x0, 0x1, 0x2729ab8, 0x0, 0x0)
        /home/mvasek/go/src/github.com/knative-sandbox/kn-plugin-func/cmd/repository.go:314 +0x15c
knative.dev/kn-plugin-func/cmd.NewRepositoryListCmd.func1(0xc000a0ea00, 0xc0008771b0, 0x0, 0x1, 0x0, 0x0)
        /home/mvasek/go/src/github.com/knative-sandbox/kn-plugin-func/cmd/repository.go:189 +0x4f
github.com/spf13/cobra.(*Command).execute(0xc000a0ea00, 0xc0008771a0, 0x1, 0x1, 0xc000a0ea00, 0xc0008771a0)
        /home/mvasek/go/src/github.com/knative-sandbox/kn-plugin-func/vendor/github.com/spf13/cobra/command.go:856 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0xc0006b6780, 0x4, 0x290b9f0, 0x3)
        /home/mvasek/go/src/github.com/knative-sandbox/kn-plugin-func/vendor/github.com/spf13/cobra/command.go:974 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /home/mvasek/go/src/github.com/knative-sandbox/kn-plugin-func/vendor/github.com/spf13/cobra/command.go:902
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        /home/mvasek/go/src/github.com/knative-sandbox/kn-plugin-func/vendor/github.com/spf13/cobra/command.go:895
main.main()
        /home/mvasek/go/src/github.com/knative-sandbox/kn-plugin-func/cmd/func/main.go:44 +0x335
```

File: repository.go (hunk 1 above)
Comment (lkingland) [reply repository.go:494]: I checked with an `https` and it worked

**Comment by lkingland** [comment]
> on actual Windows machine this even panics (it panicked even before this PR)

Yes, it seems there are several tests (8 I believe?) disabled for the time being because of something not working in Windows.  This PR unfortunately is not addressing that issue.

**Comment by matejvasek** [comment]
>Yes, it seems there are several tests (8 I believe?)

I tend to believe those are CI not general Windows issues but I cannot be sure.

**Comment by lkingland** [comment]
> if you are in a directory that is git versioned it affects output

Yes, thank you for reminding me.  That is now fixed as well.

**Comment by matejvasek** [comment]
@lkingland 
```go
if strings.HasPrefix(uri, "file://") {
		uri = r.uri[6:]
		uri = filepath.FromSlash(uri)
		uri = strings.TrimPrefix(uri, `\`)
	}
```

File: repository.go [hunk 2]
```diff
@@ -449,24 +450,60 @@ func (r *Repository) Runtime(name string) (runtime Runtime, err error) {
...
+	// The default builtin repository is indicated by an empty URI.
+	// It has no remote URL, and without this check the current working directory
+	// would be checked.
+	if uri == "" {
+		return ""
+	}
+
+	// git.PlainOpen does not seem to
+	if strings.HasPrefix(uri, "file://") {
+		uri = r.uri[6:]
```
[the remark this answers is not in the document]
Comment (matejvasek) [reply repository.go:503]: `uri = filepath.FromSlash(r.uri[7:])`

[the remark this answers is not in the document]
Comment (matejvasek) [reply repository.go:503]: That `filepath.FromSlash()` is probably not necessary but shouldn't hurt.

**Review by lance — COMMENTED** [review]
/lgtm

**Pull request merged**